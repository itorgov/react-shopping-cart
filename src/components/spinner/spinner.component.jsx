import React from 'react';

import styles from './spinner.module.scss';

export const Spinner = () => (
    <div className={styles.loader}>Loading...</div>
);
