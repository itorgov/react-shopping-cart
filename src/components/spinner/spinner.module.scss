.loader {
  color: var(--color-secondary);
  font-size: 1rem;
  margin: 1rem auto 5.5rem auto;
  position: relative;
  text-indent: -9999rem;
  transform: translateZ(0);
  animation-delay: -0.16s;

  &,
  &::before,
  &::after {
    border-radius: 50%;
    width: 2.5rem;
    height: 2.5rem;
    animation-fill-mode: both;
    animation: load 1.8s infinite ease-in-out;
  }

  &::before,
  &::after {
    content: '';
    position: absolute;
    top: 0;
  }

  &::before {
    left: -3.5rem;
    animation-delay: -0.32s;
  }

  &::after {
    left: 3.5rem;
  }
}

@keyframes load {
  0%,
  80%,
  100% {
    box-shadow: 0 2.5rem 0 -1.3rem;
  }
  40% {
    box-shadow: 0 2.5rem 0 0;
  }
}
