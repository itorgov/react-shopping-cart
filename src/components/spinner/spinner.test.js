import React from "react";
import {shallow} from "enzyme";

import {Spinner} from './spinner.component';

describe('<Spinner />', () => {
    it('expect to render', () => {
        expect(shallow(<Spinner />)).toMatchSnapshot();
    });
});
