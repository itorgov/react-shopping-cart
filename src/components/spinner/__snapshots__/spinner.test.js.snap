// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> expect to render 1`] = `
<div
  className="loader"
>
  Loading...
</div>
`;
